keepassxc-browser/background/keepass.js

```diff
@@ -409,7 +409,9 @@ keepass.associate = function(callback, tab) {
                     keepass.handleError(tab, kpErrors.ASSOCIATION_FAILED);
                 }
                 else {
-                    keepass.setCryptoKey(id, idKey);    // Save the new identification public key as id key for the database
+                    // Use public key as identification key with older KeePassXC releases
+                    const savedKey = keepass.currentKeePassXC.versionParsed > 231 ? idKey : key;
```

Comment:
Just a general remark: how is this supposed to work if we ever have a patch release with two digits?

Reply:
Not good. It would return `2.4.10` as `2410` (and a possible `2.4.0.1` as `2401`).

Reply:
And 2.41.0 would be the same as 2.4.10, right? This either needs a dedicated compare function or you need to pad with zeros up the point that it becomes unlikely that we will ever have a version with more digits (e.g., 2004000 for 2.4.0).

Reply:
Or then make an object that holds major, minor, patch and build. For this fix you can actually use that numeric compare because no version will go under it. `keepass.currentKeePassXC` also holds `version` variable which is the textual form. In this fix that could be used too with just checking `"2.3.0"` and `"2.3.1"`. The minimum version accepted by the extension is 2.3.0 anyway.